include/grpcpp/impl/codegen/proto_utils.h

```diff
@@ -76,8 +76,16 @@ Status GenericDeserialize(ByteBuffer* buffer, grpc::protobuf::Message* msg) {
   if (buffer == nullptr) {
     return Status(StatusCode::INTERNAL, "No payload");
   }
-  Status result = g_core_codegen_interface->ok();
-  {
+  Status result;
+  if (buffer->Length() < 1024 &&
+      buffer->c_buffer()->data.raw.slice_buffer.count == 1) {
```

Comment:
You already know the buffer and slice. I think there is no need to do a dump and you can use the underlying slice directly.